### DISCUSSION TITLE
SetSite failed for package [CopilotPackage]

### DISCUSSION THREAD

**Discussion by DavidAPearce** [community/community#57539] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have installed copilot 1.84.1 successfully and have a registered license but I get the following error every time I start VS (2022 v 17.6.2) Please advise aas to how to clear this and use coPilot. Thank you

SetSite failed for package [CopilotPackage]Source: &apos;Microsoft.VisualStudio.Composition&apos; 
	Description: Expected 1 export(s) with contract name &quot;GitHub.Copilot.CopilotServiceFactory&quot; but found 0 after applying applicable constraints.&#x000D;&#x000A;Microsoft.VisualStudio.Composition.CompositionFailedException: Expected 1 export(s) with contract name &quot;GitHub.Copilot.CopilotServiceFactory&quot; but found 0 after applying applicable constraints.&#x000D;&#x000A;   at Microsoft.VisualStudio.Composition.ExportProvider.GetExports(ImportDefinition importDefinition)&#x000D;&#x000A;   at Microsoft.VisualStudio.Composition.ExportProvider.GetExports[T,TMetadataView](String contractName, ImportCardinality cardinality)&#x000D;&#x000A;   at Microsoft.VisualStudio.Composition.ExportProvider.GetExport[T,TMetadataView](String contractName)&#x000D;&#x000A;   at Microsoft.VisualStudio.Composition.ExportProvider.GetExportedValue[T]()&#x000D;&#x000A;   at Microsoft.VisualStudio.ComponentModelHost.ComponentModel.GetService[T]()&#x000D;&#x000A;   at GitHub.Copilot.CopilotPackage.&lt;InitializeAsync&gt;d__5.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#x000D;&#x000A;   at Microsoft.VisualStudio.Shell.AsyncPackage.&lt;&gt;c__DisplayClass21_0.&lt;&lt;Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize&gt;b__1&gt;d.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()&#x000D;&#x000A;   at Microsoft.VisualStudio.Services.VsTask.RethrowException(AggregateException e)&#x000D;&#x000A;   at Microsoft.VisualStudio.Services.VsTask.InternalGetResult(Boolean ignoreUIThreadCheck)</description>

**Comment by lrautenbach** [comment]
Updated Visual Studio to Version 17.6.3 and now I'm getting the same error.
SetSite failed for package [CopilotPackage]Source: 'Microsoft.VisualStudio.Composition' Description: Expected 1 export(s) with contract name "GitHub.Copilot.CopilotServiceFactory" but found 0 after applying applicable constraints.
Microsoft.VisualStudio.Composition.CompositionFailedException: Expected 1 export(s) with contract name "GitHub.Copilot.CopilotServiceFactory" but found 0 after applying applicable constraints.
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExports(ImportDefinition importDefinition)
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExports[T,TMetadataView](String contractName, ImportCardinality cardinality)
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExport[T,TMetadataView](String contractName)
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExportedValue[T]()
   at Microsoft.VisualStudio.ComponentModelHost.ComponentModel.GetService[T]()
   at GitHub.Copilot.CopilotPackage.<InitializeAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.AsyncPackage.<>c__DisplayClass21_0.<<Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.Services.VsTask.RethrowException(AggregateException e)
   at Microsoft.VisualStudio.Services.VsTask.InternalGetResult(Boolean ignoreUIThreadCheck)

**Comment by ZGGSONG** [comment, marked as answer]
SetSite failed for package [CopilotPackage]Source: 'Microsoft.VisualStudio.Composition' Description: 应用适当的约束后，预期的 1 输出合同名称"GitHub.Copilot.ICopilotClientManager"但找到的 0。
Microsoft.VisualStudio.Composition.CompositionFailedException: 应用适当的约束后，预期的 1 输出合同名称"GitHub.Copilot.ICopilotClientManager"但找到的 0。
   在 Microsoft.VisualStudio.Composition.ExportProvider.GetExports(ImportDefinition importDefinition)
   在 Microsoft.VisualStudio.Composition.ExportProvider.GetExports[T,TMetadataView](String contractName, ImportCardinality cardinality)
   在 Microsoft.VisualStudio.Composition.ExportProvider.GetExport[T,TMetadataView](String contractName)
   在 Microsoft.VisualStudio.Composition.ExportProvider.GetExportedValue[T]()
   在 Microsoft.VisualStudio.ComponentModelHost.ComponentModel.GetService[T]()
   在 GitHub.Copilot.CopilotPackage.<InitializeAsync>d__3.MoveNext() 位置 /_/src/Vsix/CopilotPackage.cs:行号 52
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.VisualStudio.Shell.AsyncPackage.<>c__DisplayClass21_0.<<Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize>b__1>d.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 Microsoft.VisualStudio.Services.VsTask.RethrowException(AggregateException e)
   在 Microsoft.VisualStudio.Services.VsTask.InternalGetResult(Boolean ignoreUIThreadCheck)

**Reply by ZGGSONG** [reply]
The error disappeared after I restarted vs by typing the following command

```shell
devenv.exe /clearcache
```

**Reply by DavidAPearce** [reply]
Many thanks, Song. This is strange because after a week the error just disappeared. So it is possible that there was a clearing of the cache  done automatically at some point
